rainier-core/src/main/scala/com/stripe/rainier/core/Categorical.scala

```diff
@@ -24,8 +25,14 @@ final case class Categorical[T](pmf: Map[T, Real]) extends Distribution[T] {
         .map { case (u, ups) => u -> Real.sum(ups.map(_._2)) }
         .toMap)
 
-  def logDensity(t: T): Real =
-    pmf.getOrElse(t, Real.zero).log
+  //there should be exactly one value set to 1.0
```

Comment:
So rather than computing `logDensity` of a `t: T`, we compute `logDensity` of a `Map[T, Real]` of the form `Map(t -> Rea.one)`, right?

Reply:
Gotcha.